internal/integration/infra_sdk_emitter_test.go

```diff
@@ -380,33 +306,165 @@ redis_foo_test{hostname="localhost",env="dev",uniquelabel="test"} 3
+							"targetName":        "128.0.0.1:8080",
+							"scrapedTargetName": "128.0.0.1:8080",
+							"scrapedTargetURL":  "https://128.0.0.1:8080",
+							"env":               "dev",
+						},
+					},
+				},
+				Target: endpoints.Target{},
+			},
+			expectedName: "128.0.0.1:8080",
+			expectedURL:  "https://128.0.0.1:8080",
+			hostID:       "a-host-id",
+		},
+		{
+			testName: "provided host id modifying name if localhost",
+			input: TargetMetrics{
+				Metrics: []Metric{
+					{
+						name:       "a-metric",
+						value:      float64(3),
+						metricType: "count",
+						attributes: labels.Set{
+							"targetName":        "localhost:8080",
+							"scrapedTargetName": "localhost:8080",
+							"scrapedTargetURL":  "https://localhost:8080",
+							"env":               "dev",
+						},
+					},
+				},
+				Target: endpoints.Target{},
+			},
+			expectedName: "a-host-id:8080",
+			expectedURL:  "https://localhost:8080",
+			hostID:       "a-host-id",
+		},
+		{
+			testName: "empty host id modifying name if LOCALHOST",
+			input: TargetMetrics{
+				Metrics: []Metric{
+					{
+						name:       "a-metric",
+						value:      float64(3),
+						metricType: "count",
+						attributes: labels.Set{
+							"targetName":        "LOCALHOST:8080",
+							"scrapedTargetName": "LOCALHOST:8080",
+							"scrapedTargetURL":  "https://LOCALHOST:8080",
+							"env":               "dev",
+						},
+					},
+				},
+				Target: endpoints.Target{},
+			},
+			expectedName: "a-host-id:8080",
+			expectedURL:  "https://LOCALHOST:8080",
+			hostID:       "a-host-id",
+		},
+		{
+			testName: "empty host id modifying name if 127.0.0.1",
+			input: TargetMetrics{
+				Metrics: []Metric{
+					{
+						name:       "a-metric",
+						value:      float64(3),
+						metricType: "count",
+						attributes: labels.Set{
+							"targetName":        "127.0.0.1:8080",
+							"scrapedTargetName": "127.0.0.1:8080",
+							"scrapedTargetURL":  "https://127.0.0.1:8080",
+							"env":               "dev",
+						},
+					},
+				},
+				Target: endpoints.Target{},
+			},
+			expectedName: "a-host-id:8080",
+			expectedURL:  "https://127.0.0.1:8080",
+			hostID:       "a-host-id",
+		},
+		{
+			testName: "empty host id not modifying if empty",
+			input: TargetMetrics{
+				Metrics: []Metric{
+					{
+						name:       "a-metric",
+						value:      float64(3),
+						metricType: "count",
+						attributes: labels.Set{
+							"targetName":        "localhost:8080",
+							"scrapedTargetName": "localhost:8080",
+							"scrapedTargetURL":  "https://localhost:8080",
+							"env":               "dev",
+						},
+					},
+				},
+				Target: endpoints.Target{},
+			},
+			expectedName: "localhost:8080",
+			expectedURL:  "https://localhost:8080",
+			hostID:       "",
+		},
+	}
+
+	testMetadata := Metadata{
+		Name:    "nri-foo",
+		Version: "test",
+	}
+
+	for _, c := range cases {
+		c := c
+
+		t.Run(c.testName, func(t *testing.T) {
+
```

Comment:
not actually needed but if possible t.parallel should be used.

Reply:
t.Parallel() is intended to be used for slow tests only; average tests are so fast that any gain from parallel execution would be negligible. I really don't know why we are using it everywhere if go test -race is used.

Reply:
It actually tends to mess up table driven tests adding extra complexity for no reason.